app/views/RoomActionsView/index.tsx

```diff
@@ -139,15 +147,33 @@ class RoomActionsView extends React.Component<IRoomActionsViewProps, IRoomAction
 		this.mounted = true;
 		const { room, member } = this.state;
 		if (room.rid) {
-			if (!room.id && !this.isOmnichannelPreview) {
-				try {
-					const result = await RocketChat.getChannelInfo(room.rid);
-					if (result.success) {
-						// @ts-ignore
-						this.setState({ room: { ...result.channel, rid: result.channel._id } });
+			// there is room.id for rooms in db
+			if (!room.id) {
+				// need to check if the type is from omnichannel
+				if (room.t === SubscriptionType.OMNICHANNEL) {
+					// if is a Preview, we don't need to do anything
+					// however if isn't a Preview, the room should be in db
+					if (!this.isOmnichannelPreview) {
+						const db = database.active;
+						const subCollection = db.get('subscriptions');
+						try {
+							const result = await subCollection.find(room.rid);
```

Comment:
Can you use `getSubscriptionByRoomId`?
It's going to clean up this code a little bit.